### PR TITLE
Adding Port to tenant resolution

### PR DESCRIPTION
**Pull request by Jetski5822**
at the moment if you have two tenants...

1. Default
2. one with a host of localhost:10000

localhost:10000 will not be resolved, and will fall back to default.

### CODE REVIEW

File: src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs [hunk 1]
```diff
@@ -100,24 +100,38 @@ public ShellSettings Match(string host, string appRelativePath)
         private bool TryMatchInternal(string host, string appRelativePath, out ShellSettings result)
         {
             // 1. Search for Host + Prefix match
-            string hostAndPrefix = GetHostAndPrefix(host, appRelativePath);
+            var hostAndPrefix = GetHostAndPrefix(host, appRelativePath);
 
             if (!_shellsByHostAndPrefix.TryGetValue(hostAndPrefix, out result))
             {
-                // 2. Search for Host only match
+                // 2. Search for Full Host only match
```
Comment (sebastienros) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: Why "Full"? How is the port influencing the selection?

**Comment by sebastienros** [comment]
So the issue is that one of the tenants don't specify a host, and gets selected unlike the one that has a matching host?

File: test/Orchard.Tests/Shell/RunningShellTableTests.cs [hunk 2]
```diff
@@ -101,6 +101,16 @@ public void DefaultCatchallIsFallbackEvenWhenOthersAreUnqualified()
         }
 
         [Fact]
+        public void doport()
```
Comment (sebastienros) [inline test/Orchard.Tests/Shell/RunningShellTableTests.cs]: Name?

**Comment by Jetski5822** [comment]
No, the port gets ignored during resolution.

File: test/Orchard.Tests/Shell/RunningShellTableTests.cs (hunk 2 above)
Comment (Jetski5822) [inline test/Orchard.Tests/Shell/RunningShellTableTests.cs]: opps.

File: src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs (hunk 1 above)
Comment (sebastienros) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: I mean we never have had to do anything for ports, why should we now?

Comment (Jetski5822) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: Its something that came up where I am working... and why not? Its a valid use case.

They would rather use ports than virtual paths or host entries.

Comment (sebastienros) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: This is wrong, on a server all the ports are the same, hence any value should work.

File: test/Orchard.Tests/Shell/RunningShellTableTests.cs (hunk 2 above)
Comment (Jetski5822) [inline test/Orchard.Tests/Shell/RunningShellTableTests.cs]: fixed.

File: src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs (hunk 1 above)
Comment (sebastienros) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: I mean on a multi-tenant orchard instance, there is one server, so one port

Comment (sebastienros) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: What I see here is a mis-configuation of your tenants, you set localhost where actually you should not have any host specified.

**Comment by Jetski5822** [comment]
not needed as we are fools.

**Pull request closed without merging**